scarb/src/compiler/compilers/executable.rs

```diff
@@ -71,3 +82,114 @@ impl Compiler for ExecutableCompiler {
         )
     }
 }
+
+fn compile_executable(
+    unit: &CairoCompilationUnit,
+    db: &RootDatabase,
+    ws: &Workspace<'_>,
+    executable_path: Option<&str>,
+    main_crate_ids: Vec<CrateId>,
+    mut diagnostics_reporter: DiagnosticsReporter<'_>,
+    config: ExecutableConfig,
+) -> Result<CompiledFunction> {
+    let executables = find_executable_functions(db, main_crate_ids, executable_path);
+
+    let executable = match executables.len() {
+        0 => {
+            // Report diagnostics as they might reveal the reason why no executable was found.
+            diagnostics_reporter.ensure(db)?;
+            anyhow::bail!("Requested `#[executable]` not found.");
+        }
+        1 => executables[0],
+        _ => {
+            let executable_names = executables
+                .iter()
+                .map(|executable| originating_function_path(db, *executable))
+                .sorted()
+                .collect_vec();
+            let scarb_toml = unit
+                .main_component()
+                .package
+                .manifest_path()
+                .workspace_relative(ws);
+            anyhow::bail!(ambiguous_executable_error_message(
+                executable_names,
+                scarb_toml
+            ));
+        }
+    };
+
+    compile_executable_function_in_prepared_db(db, executable, diagnostics_reporter, config)
+}
+
+fn ambiguous_executable_error_message(executables: Vec<String>, scarb_toml: &Utf8Path) -> String {
```

Comment:
I feel like current name may be misleading 

```suggestion
fn multiple_executables_error_message(executables: Vec<String>, scarb_toml: &Utf8Path) -> String {
```